src/ln/channelmanager.rs

```diff
@@ -240,6 +241,8 @@ pub struct ChannelManager {
 	channel_state: Mutex<ChannelHolder>,
 	our_network_key: SecretKey,
 
+	unsolved_htlcs: Mutex<Vec<[u8;32]>>,
```

Comment:
Unsolved htlc is a subset of claimable_htlcs as we can't iter on them at the same time and call downstream claim_funds/fail_backwards with another lock on it.

Reply:
Why not move the HTLCPreviousHopData into the ChannelMonitor when the Channel force-closes? I'd be kinda OK with breaking privacy and doing a HashMap<[u8; 32], HTLCPreviousHopData> there because other similar issues are gonna exist anyway (see #168).

Reply:
Nah, I succeed to get around lock issues in ChannelManager block_connected, so I can get ride of unsolved_htlcs (in fact, real names should be unresolved htlc accordingly to BOLT 5..). Hopefully, will finish the other parts of PR tomorrow

Reply:
The current patch doesn't appear to *use* unsolved_htlcs for anything? It's only ever written to.

Reply:
Mistake of mine, forget to clean it, should be gone as aed435a